feat: access project access control for invite modal

zlwaterfield opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changes

We support project access controls to be created upon invite acceptance. It's currently only supported via the API. This PR is adding it to the invite modal so users can start to use it. It's behind a feature flag for internal testing.

How it works:
- when the invite modal is opened, it loads all projects (team) on the organization
- if the user has the feature flag and advanced permissions (meaning they can set access controls), they will see a dropdown to select projects they want to provide access to
- when a project is selected it loads the access controls for that project to show the default access level and get info for the banners
- they can select member or admin, that will add an access control row for the invited upon acceptance for the project with that level
- there are warning banners in place if the invite level doesn't make sense based on other factors (i.e. you're inviting an admin)

Notes
- I chose to allow all projects and not filter by those that have default access controls in order to reduce confusion
- this won't be a great UI if the user have a ton of projects or is inviting a bunch of users at once - we can iterate over time.
- I am looking for feedback on the UI if you have any

<img width="825" height="791" alt="Screenshot 2025-07-15 at 10 44 06 AM" src="https://github.com/user-attachments/assets/d8108031-44c1-48fb-82fc-712d4e396d1c" />

<img width="832" height="555" alt="Screenshot 2025-07-15 at 3 06 17 PM" src="https://github.com/user-attachments/assets/b0f12bbc-d530-438c-8dc5-f7d190ca9140" />

<img width="347" height="209" alt="Screenshot 2025-07-15 at 3 06 35 PM" src="https://github.com/user-attachments/assets/81ae9bf4-c169-41d9-9828-a5f7aef303ab" />

<img width="428" height="145" alt="Screenshot 2025-07-15 at 3 17 48 PM" src="https://github.com/user-attachments/assets/6963627e-777f-4b77-8a6e-accd6b7518db" />

Here is the UI before:

<img width="812" height="494" alt="Screenshot 2025-07-15 at 4 04 34 PM" src="https://github.com/user-attachments/assets/ac79d938-3c74-4a84-91d9-221e38679f95" />


## Did you write or update any docs for this change?

<!-- Engineers are responsible for doing the first pass at documenting their features and/or code.  -->

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [ ] No docs needed for this change

## How did you test this code?

Manually

zlwaterfield committed:
access project access control for invite modal

zlwaterfield committed:
LINT

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

zlwaterfield committed:
fix team resource

zlwaterfield committed:
Update InviteModal.tsx

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

joshsny commented on frontend/src/scenes/settings/organization/inviteLogic.ts:
```diff
@@ -65,6 +75,29 @@ export const inviteLogic = kea<inviteLogicType>([
+            {
+                loadProjectAccessControl: async (projectId: number) => {
+                    try {
+                        const accessControls = await api.get(`api/projects/${projectId}/access_controls`)
```
Does this API support filtering by access control type, if not do we have a page limit on it that might cut off finding the project access control?

zlwaterfield replied:
We don't but I plan to add filtering and then will add it here.

joshsny reviewed (approved):
Low on context with these so apologies in advance for some irrelevant comments!

zlwaterfield committed:
Update frontend/src/scenes/settings/organization/InviteModal.tsx

Co-authored-by: Joshua Snyder <joshua@posthog.com>

zlwaterfield commented:
>

> Nice! Had a couple questions on the UI:
> 
> * I imagine people who care about this stuff generally have lots of invites to make, do we have plans for a bulk experience for this (e.g. pasting users auto-fills the modal or something like that)? (obvs out of scope of this PR)
> * The warning message in the first screen is not clear to me. Why can't I invite someone as a member if the default is admin? Is admin the minimum role?

1. Yes, that is something we can look into adding but actually hasn't been requested.
2. Thanks for the heads up, I'll iterate on it.

zlwaterfield commented on frontend/src/scenes/settings/organization/InviteModal.tsx:
```diff
@@ -38,9 +40,147 @@ export function EmailUnavailableMessage(): JSX.Element {
+                                            {
+                                                value: TeamMembershipLevel.Member,
+                                                label: TeamMembershipLevel[TeamMembershipLevel.Member],
+                                            },
```
enum are weird

This enum
```
export enum TeamMembershipLevel {
    Member = 1,
    Admin = 8,
}
```
results in this: 
```
// TeamMembershipLevel
{1: 'Member', 8: 'Admin', Member: 1, Admin: 8}
// TeamMembershipLevel.Admin
8
// TeamMembershipLevel.Member
1
// TeamMembershipLevel[TeamMembershipLevel.Admin]
Admin
//TeamMembershipLevel[TeamMembershipLevel.Member]
Member
```

zlwaterfield committed:
Update type for private_project_access level

Changed the type of the 'level' property in OrganizationInviteType.private_project_access to use AccessControlLevel enum values instead of string literals for better type safety and consistency.

zlwaterfield commented:
@joshsny, made the banner more clear
<img width="863" height="474" alt="Screenshot 2025-07-16 at 4 58 13 PM" src="https://github.com/user-attachments/assets/edb3866c-c2c9-43dc-80e0-e55385a4af0f" />

zlwaterfield committed:
improve types and copy

raquelmsmith commented on frontend/src/scenes/settings/organization/InviteModal.tsx:
```diff
@@ -38,9 +40,160 @@ export function EmailUnavailableMessage(): JSX.Element {
+                                <div className="p-2 bg-bg-light rounded border">
+                                    {isLowerThanDefault && (
+                                        <div className="mb-2">
+                                            <LemonBanner type="warning" className="text-xs">
+                                                <strong>{project.name}</strong> has a default access level of{' '}
+                                                <span className="font-bold italic">{defaultLevel}</span>. Since you
+                                                selected <span className="font-bold italic">{access.level}</span> (which
+                                                is lower), the user will actually get{' '}
+                                                <span className="font-bold italic">{defaultLevel}</span> access.
+                                            </LemonBanner>
```
IMO this is kind of silly, shouldn't we prevent them from selecting a membership level lower than the default? Show it in the level dropdown, but not selectable, and a tooltip saying why they can't select it.

raquelmsmith reviewed (commented):
This is awesome, thank you for finishing up the work I never finished 😄

zlwaterfield committed:
update the checks for the invites

zlwaterfield committed:
Disable invite actions while inviting team members

Added an isInviting state to prevent multiple invite submissions and disable the Cancel button while invites are being processed in the InviteModal.

zlwaterfield committed:
Support both int and str access levels in invites

Update organization invite logic to accept both legacy integer and new string-based access levels for project access. This ensures compatibility during the transition to the new access control system.

zlwaterfield committed:
Merge remote-tracking branch 'origin/master' into zach/invites-private-projects

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

zlwaterfield committed:
LINT
